### PR TITLE
Move tenant buttons into dropdown menu

### PR DESCRIPTION
**Pull request by hishamco**
Address a part of #12006 

https://user-images.githubusercontent.com/3237266/179314511-f90a6757-2440-4787-950e-54b07bb94fc1.mp4

### CODE REVIEW

**Comment by Skrypt** [comment]
You are missing this : `id="btn-setup-@entry.Name"`
On the setup button.

**Comment by Skrypt** [comment]
You will need to add a click trigger on the dropdown menu before the click on the setup button.

**Comment by Skrypt** [comment]
The functional tests are not executed under Windows that's why.

File: src/OrchardCore.Modules/OrchardCore.Tenants/Views/Admin/Index.cshtml [hunk 1]
```diff
@@ -146,25 +146,29 @@
...
-                                    <a asp-action="Enable" asp-route-id="@entry.Name" class="btn btn-success btn-sm enable" data-url-af="UnsafeUrl">@T["Enable"]</a>
-                                }
-                                @if (entry.ShellSettings.State == TenantState.Uninitialized)
-                                {
-                                    <a class="btn btn-info btn-sm setup" id="btn-setup-@entry.Name" href="@GetEncodedUrl(entry, originalPathBase)">@T["Setup"]</a>
-                                }
-                                <a asp-action="Reload" asp-route-id="@entry.Name" class="btn btn-secondary btn-sm reload" data-url-af="UnsafeUrl">@T["Reload"]</a>
+                            <div class="dropdown float-end">
+                                <a id="tenantDropdownMenuButton" class="btn text-secondary" data-bs-toggle="dropdown" aria-expanded="false"><i class="fa fa-ellipsis-v"></i></a>
+                                <ul class="dropdown-menu" aria-labelledby="tenantDropdownMenuButton">
```
Comment (ns8482e) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Views/Admin/Index.cshtml:151]: aria-labelledby Ref doesn’t have any text

**Comment by hishamco** [comment]
The states are clearly shown as a badge. The current UI change:

- Makes a lot of room in each row
- Makes the UI clean & clear
- The actions should react based on some state
- No more colored buttons which are sometime confuses

FYI such menu actions already shown in GitHub & AzureDevOps for better UX. May be the issue we are looking from developer point of you ;)

**Comment by Skrypt** [comment]
Ok, we need to make sure we are consistent here. While I don't dislike entirely the UI change; here we need to take into consideration that the Content Items list uses an "Action" button for these. This is why I recommended using the same UI concept.
I'm sure DevOps uses also single item buttons in some places.

**Comment by hishamco** [comment]
Actions button

**Comment by ns8482e** [comment]
> Actions button

Personally I don’t like devops UI,  I like GitHub’s UI better

**Comment by hishamco** [comment]
GitHub

It quite similar except the action button uses dropdown menu similar to what we have in filters

**Comment by Skrypt** [comment]
The inline action buttons are simply icons here on Github.

**Comment by MikeAlhayek** [comment]
@hishamco thanks for putting together this PR. Can you please fix the conflict? 

I personally do not like hiding all the buttons for all actions. Very common functions like "Edit" should be visible all the time. I don't want to have to make 2 click to perform a common functions. The idea here is improve the UI not complicating it like devops :). I am also not a big fan of DevOps UI as @ns8482e mentioned. 


I think in this UI, we should show "Edit" and "View" buttons, then move the "Reload", "Enable", "Disable", "Features", and "Remove" button should be in the "Actions" button.

**Comment by hishamco** [comment]
> @hishamco thanks for putting together this PR. Can you please fix the conflict?

Sure

>  personally do not like hiding all the buttons for all actions. Very common functions like "Edit" should be visible all the time

May be but the issue is everyone of us has it's own common functionality ;)

**Comment by hishamco** [comment]
Let me check this with a UI expert and see what I can come up with

**Comment by Piedone** [comment]
Putting the buttons under a nondescript three-dot menu is a hard no from me.

**Comment by MikeAlhayek** [comment]
... is a hard no for me to. I am okay with "Actions" drop down button if that is an option

**Comment by hishamco** [comment]
> Putting the buttons under a nondescript three-dot menu is a hard no from me.

It's much better for UI clarity

**Comment by Piedone** [comment]
How does hiding buttons clarify intent?

**Comment by hishamco** [comment]
From UI/UX perspective when there are many buttons like what we have, it would be nice to introduce an action menu instead

**Comment by Piedone** [comment]
This is how the current UI looks like if all the possible buttons are displayed, in a common full HD resolution:

image

Ultimately, design is subjective until we do large sample size research for this with focus groups (which would probably be excessive for this) but this doesn't seem to be too many buttons for me. Nor there is any lack of screen real estate that we'd use better. Same if you have 10 tenants on a page.

Did you encounter somebody complaining about this UI?

**Comment by hishamco** [comment]
> Did you encounter somebody complaining about this UI?

While I saw many web framework applications, & CMSs I never see something like this :) Also the coloring is another story lol

**Comment by Piedone** [comment]
Do you want to bring this up at the meeting or something so we can get a closure? Here we have 2-3 against votes, and other remarks that necessitate changes even if the basic direction is agreed on.

**Comment by hishamco** [comment]
Probably yes, I have this and Culture Picker in localization part #13354

**Comment by MikeAlhayek** [comment]
@hishamco @Piedone I think we should have 3 buttons are we do in contents. The buttons should be View, Edit and Actions (drop down menu). But I do not like the ... button.

This way we can simplify all the buttons we have today and keep them consistent with other pages like controller.

**Comment by Piedone** [comment]
I'm OK with that.

**Comment by hishamco** [comment]
> But I do not like the ... button

Sure everyone will like and dislike some of the UI, but I was using things from UI/UX perspective

**Comment by hishamco** [comment]
I will close this while there's not much interest in UI improvements, so I will move this to Orchard Core Contrib (OCC), while I'm already some improvements to the admin UI

**Pull request closed without merging**